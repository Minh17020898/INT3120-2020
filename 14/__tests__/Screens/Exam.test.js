import React from 'react';
import renderer from 'react-test-renderer';
import Exam from '../../Screens/Exam';

describe('test exam', () => {
  const exam = renderer.create(
  <Exam 
    navigation = {
      {
        navigate: jest.fn(),
        state: {
            params: {
                type: 'Part II'
            }
        }
      }
    }
  />);
  const examJSON = exam.toJSON();
  const examInstance = exam.getInstance();
  // test 1
  it('renders correctly', () => {
    expect(examJSON).toMatchSnapshot();
  });
})
