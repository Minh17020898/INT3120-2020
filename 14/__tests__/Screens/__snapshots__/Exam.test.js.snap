// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test exam renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#d2d6d9",
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2089db",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "height": 64,
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "paddingTop": 20,
      }
    }
  >
    <Image
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 64,
            "width": undefined,
          },
          undefined,
        ]
      }
      testID="headerContainer"
    />
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 3,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "#fff",
          }
        }
        text="Part II"
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        Part II
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
        }
      }
    />
  </View>
  <RCTScrollView
    data={
      Array [
        Object {
          "answer1": "f",
          "answer2": "t",
          "answer3": "f",
          "category": "Part I",
          "favorite": 0,
          "question": "test 1",
          "result": "two",
        },
        Object {
          "answer1": "t",
          "answer2": "f",
          "answer3": "f",
          "category": "Part II",
          "favorite": 0,
          "question": "test 2",
          "result": "one",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "opacity": 0,
              "transform": Array [
                Object {
                  "translateX": -100,
                },
              ],
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#d2d6d9",
                "borderBottomWidth": 1,
                "borderColor": "#000000",
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#ffffff",
                  "borderColor": "black",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "marginLeft": 16,
                  "marginRight": 16,
                  "paddingBottom": 8,
                  "paddingLeft": 8,
                  "paddingTop": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#000000",
                  }
                }
              >
                test 1
              </Text>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "margin": 5,
                      "marginLeft": 10,
                      "marginRight": 10,
                      "opacity": 1,
                      "padding": 10,
                    }
                  }
                  testID="checkbox"
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "marginTop": 18,
                      },
                      null,
                    ]
                  }
                >
                  f
                </Text>
              </View>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "margin": 5,
                      "marginLeft": 10,
                      "marginRight": 10,
                      "opacity": 1,
                      "padding": 10,
                    }
                  }
                  testID="checkbox"
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "marginTop": 18,
                      },
                      null,
                    ]
                  }
                >
                  t
                </Text>
              </View>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "margin": 5,
                      "marginLeft": 10,
                      "marginRight": 10,
                      "opacity": 1,
                      "padding": 10,
                    }
                  }
                  testID="checkbox"
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "marginTop": 18,
                      },
                      null,
                    ]
                  }
                >
                  f
                </Text>
              </View>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "flex-end",
                    "width": "100%",
                  }
                }
              >
                <View>
                  <View
                    accessible={true}
                    focusable={true}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
                <View>
                  <View
                    accessible={true}
                    focusable={true}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </View>
            <RNGestureHandlerButton
              collapsable={false}
              onGestureEvent={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              onHandlerStateChange={[Function]}
              rippleColor={0}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  undefined,
                ]
              }
            >
              <View
                accessible={true}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                    theme={
                      Object {
                        "colors": Object {
                          "disabled": "hsl(208, 8%, 90%)",
                          "divider": "#bcbbc1",
                          "error": "#ff190c",
                          "grey0": "#393e42",
                          "grey1": "#43484d",
                          "grey2": "#5e6977",
                          "grey3": "#86939e",
                          "grey4": "#bdc6cf",
                          "grey5": "#e1e8ee",
                          "greyOutline": "#bbb",
                          "platform": Object {
                            "android": Object {
                              "error": "#f44336",
                              "primary": "#2196f3",
                              "secondary": "#9C27B0",
                              "success": "#4caf50",
                              "warning": "#ffeb3b",
                            },
                            "ios": Object {
                              "error": "#ff3b30",
                              "primary": "#007aff",
                              "secondary": "#5856d6",
                              "success": "#4cd964",
                              "warning": "#ffcc00",
                            },
                          },
                          "primary": "#2089dc",
                          "searchBg": "#303337",
                          "secondary": "#8F0CE8",
                          "success": "#52c41a",
                          "warning": "#faad14",
                        },
                      }
                    }
                    underlayColor="white"
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </RNGestureHandlerButton>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "opacity": 0,
              "transform": Array [
                Object {
                  "translateX": -100,
                },
              ],
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#d2d6d9",
                "borderBottomWidth": 1,
                "borderColor": "#000000",
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#ffffff",
                  "borderColor": "black",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "marginLeft": 16,
                  "marginRight": 16,
                  "paddingBottom": 8,
                  "paddingLeft": 8,
                  "paddingTop": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#000000",
                  }
                }
              >
                test 2
              </Text>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "margin": 5,
                      "marginLeft": 10,
                      "marginRight": 10,
                      "opacity": 1,
                      "padding": 10,
                    }
                  }
                  testID="checkbox"
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "marginTop": 18,
                      },
                      null,
                    ]
                  }
                >
                  t
                </Text>
              </View>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "margin": 5,
                      "marginLeft": 10,
                      "marginRight": 10,
                      "opacity": 1,
                      "padding": 10,
                    }
                  }
                  testID="checkbox"
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "marginTop": 18,
                      },
                      null,
                    ]
                  }
                >
                  f
                </Text>
              </View>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "margin": 5,
                      "marginLeft": 10,
                      "marginRight": 10,
                      "opacity": 1,
                      "padding": 10,
                    }
                  }
                  testID="checkbox"
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "marginTop": 18,
                      },
                      null,
                    ]
                  }
                >
                  f
                </Text>
              </View>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "flex-end",
                    "width": "100%",
                  }
                }
              >
                <View>
                  <View
                    accessible={true}
                    focusable={true}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
                <View>
                  <View
                    accessible={true}
                    focusable={true}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </View>
            <RNGestureHandlerButton
              collapsable={false}
              onGestureEvent={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              onHandlerStateChange={[Function]}
              rippleColor={0}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  undefined,
                ]
              }
            >
              <View
                accessible={true}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                    theme={
                      Object {
                        "colors": Object {
                          "disabled": "hsl(208, 8%, 90%)",
                          "divider": "#bcbbc1",
                          "error": "#ff190c",
                          "grey0": "#393e42",
                          "grey1": "#43484d",
                          "grey2": "#5e6977",
                          "grey3": "#86939e",
                          "grey4": "#bdc6cf",
                          "grey5": "#e1e8ee",
                          "greyOutline": "#bbb",
                          "platform": Object {
                            "android": Object {
                              "error": "#f44336",
                              "primary": "#2196f3",
                              "secondary": "#9C27B0",
                              "success": "#4caf50",
                              "warning": "#ffeb3b",
                            },
                            "ios": Object {
                              "error": "#ff3b30",
                              "primary": "#007aff",
                              "secondary": "#5856d6",
                              "success": "#4cd964",
                              "warning": "#ffcc00",
                            },
                          },
                          "primary": "#2089dc",
                          "searchBg": "#303337",
                          "secondary": "#8F0CE8",
                          "success": "#52c41a",
                          "warning": "#faad14",
                        },
                      }
                    }
                    underlayColor="white"
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </RNGestureHandlerButton>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "marginTop": 22,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderRadius": 10,
            "elevation": 5,
            "maxHeight": 800,
            "padding": 20,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3.84,
            "width": 250,
          }
        }
      >
        <Text>
          You have completed 
          Part II
        </Text>
        <View
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              OK
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "width": "50%",
        }
      }
    />
    <View
      style={
        Object {
          "borderColor": "white",
          "borderRightWidth": 4,
          "borderWidth": 2,
          "width": "50%",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Next
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
